tests/test_survey_weights.py

```
@@ -0,0 +1,81 @@
+    def test_is_runtime_error_subclass(self):
+        err = WeightingConvergenceError("test")
+        assert isinstance(err, RuntimeError)
+
+    def test_message(self):
+        err = WeightingConvergenceError("did not converge")
+        assert "converge" in str(err)
+
+
+@pytest.mark.skipif(not WEIGHTIPY_AVAILABLE, reason="weightipy not installed")
+class TestApplyRimWeightsWithWeightipy:
+    def _sample_df(self):
```

Ruff (0.15.10)

[warning] Missing return type annotation for private function `_sample_df`

(ANN202)